### PR TITLE
Major overhaul of documentation

### PR DESCRIPTION
**Pull request by muhrin**
This should go on top of #45 .

* New template
* Logo
* Jupyter examples
* API doc update
* README updated

### CODE REVIEW

**Comment by muhrin** [comment]
Woohoo, last one!  You can see these rendered here:
https://kiwipy-maruh.readthedocs.io/en/develop/

File: docs/source/concepts.rst [hunk 1]
```diff
@@ -1,33 +1,45 @@
...
-****************
-Communicator
-****************
+Coroutines vs Threads
+=====================
 
-Loop communicator
-==================
+KiwiPy's RabbitMQ backend has both coroutine and threaded version of the Communicator.  If you're in a coroutine
+environment you may prefer to use the :class:`mincepy.rmq.RmqCommunicator` which uses coroutines and
```
Comment (sphuber) [inline docs/source/concepts.rst:40]: were you hungry when you wrote this? I agree a minced pie is delicious but I thought you were baking a kiwipy over here

File: docs/source/concepts.rst [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
+:class:`asyncio.Future` s while if you don't have an event loop just stick with the :class:`mincepy.rmq.RmqThreadCommunicator` which
```
Comment (sphuber) [inline docs/source/concepts.rst:41]: idem dito

File: docs/source/examples/broadcast.ipynb [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Broadcast received:\n",
+      "sender:\tme\n",
+      "subject:yo momma\n",
+      "body:\tso fat\n",
```
Comment (sphuber) [inline docs/source/examples/broadcast.ipynb:65]: Actually, given that we want to publish this, I think it is best to change this example. I don't care personally about the language, but it is not the classiest.

File: docs/source/concepts.rst (hunk 1 above)
Comment (muhrin) [reply docs/source/concepts.rst]: Haha, yeah, hungry for code.  Got too many pies in the oven and confused with my database library.

File: docs/source/examples/broadcast.ipynb (hunk 3 above)
Comment (muhrin) [reply docs/source/examples/broadcast.ipynb]: Aww, no fun!  Fine - changed to a Futurama reference.

**Comment by muhrin** [comment]
Yeah man, super pleased with the way it looks - Sonia put the logo together.

**Pull request merged**